roles/openshift_health_checker/openshift_checks/logging/podlogs.py

```diff
@@ -0,0 +1,174 @@
+        if error:
+            return {"failed": True, "msg": error}
+
+        return {}
+
+    def check_log_delay(self, kibana_pod, es_pod, pod_log_timeout, task_vars):
+        """Check delay of logs and return any errors accordingly"""
+        error = None
+
+        try:
+            uid = self.curl_kibana_with_uuid(kibana_pod, task_vars)
+            self.wait_until_cmd_or_err(es_pod, uid, pod_log_timeout, task_vars)
+        except OpenShiftCheckException as error:
+            error = str(error)
+        except KeyError as error:
+            error = "Invalid json response from pod: {}".format(str(error))
+        except ValueError as error:
+            error = "Invalid response from pod {}".format(str(error))
+
+        return error
+
+    @staticmethod
+    def running_pods(pods):
+        """Returns: list of pods in a running state"""
+        return [
+            pod for pod in pods
+            if pod['status']['phase'] == 'Running'
+        ]
+
+    def wait_until_cmd_or_err(self, es_pod, uid, timeout_secs, task_vars):
+        """Wait a maximum of timeout_secs for the uuid logged in Kibana to be found in
+        the Elasticsearch logs. Since we are querying for a message with the the unique
+        uuid from earlier, there should only be a single match.
+        Raise an OpenShiftCheckException if the timeout is reached before a result is found."""
+
+        interval = 1  # seconds to wait between retries
+        orig_timeout = timeout_secs
+
+        while timeout_secs > 0:
+            time.sleep(interval)
+            timeout_secs -= interval
+
+            total = self.query_es_from_es(es_pod, uid, task_vars)
+            if total == 1:
+                return
+
+        msg = "expecting match in Elasticsearch for message with uuid {}, but no matches were found after {}s."
+        raise OpenShiftCheckException(msg.format(uid, orig_timeout))
+
+    def curl_kibana_with_uuid(self, kibanna_pod, task_vars):
+        """curl Kibana with a unique uuid."""
+        uid = str(self.generate_uuid())
+        pod_name = kibanna_pod["metadata"]["name"]
+        exec_cmd = "exec {pod_name} -c kibana -- curl --connect-timeout 60 -s http://localhost:5601/{uid}"
+
+        error_str = self.oc_cmd(
+            exec_cmd.format(
+                pod_name=pod_name,
+                uid=uid
+            ),
+            [], task_vars,
+        )
+        error_code = json.loads(error_str).get("statusCode")
+
+        if error_code != 404:
+            msg = "expecting server error code 404, but got {} instead."
+            raise OpenShiftCheckException(msg.format(error_code))
+
+        return uid
+
+    def query_es_from_es(self, es_pod, uid, task_vars):
+        """curl the Elasticsearch pod and look for a unique uuid in its logs."""
+        pod_name = es_pod["metadata"]["name"]
+        exec_cmd = ("exec {pod_name} -- curl --connect-timeout 60 -s -k -f "
```

Comment:
I don't *think* there is a need for `-k` here, should be able to just specify the CA and have a valid TLS connection... remind me to check into it.